main/sources/hear-this.ts

```diff
@@ -29,46 +42,146 @@ class HearThis implements ProjectSource {
+      sourceType: this.SOURCE_TYPE,
+      books: [],
+    };
+    const bookNames = sortInCanonicalOrder(getDirectories(project.folderPath));
+    const projectBooks = bookNames
+      .map((bookName: string) => this.makeBook(projectName, bookName, directory))
+      .filter((book: BKBook) => book.chapters.length);
+    for (const book of projectBooks) {
+      project.books.push(book);
+    }
     return project;
   }
 
-  makeBook(projectName: string, name: string, directory: string): Book {
-    const book = new Book();
-    book.name = name;
-    const chapterNames = getDirectories(path.join(directory, projectName, name));
+  private makeBook(projectName: string, bookName: string, directory: string): BKBook {
+    const book: BKBook = {
+      name: bookName,
+      chapters: [],
+    };
+    const chapterNames = getDirectories(path.join(directory, projectName, bookName));
 
     const naturalSortChapterNames = chapterNames.sort(
       new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' }).compare
     );
 
-    book.chapters = naturalSortChapterNames
-      .map((chapterName: string) => this.makeChapter(projectName, name, chapterName, directory))
-      .filter((chapter: Chapter) => chapter.audioFiles.length > 0);
+    const bookChapters = naturalSortChapterNames
+      .map((chapterName: string) => this.makeChapter(projectName, bookName, chapterName, directory))
+      .filter((chapter: BKChapter) => chapter.audio.files.length);
+    for (const chapter of bookChapters) {
+      book.chapters.push(chapter);
+    }
     return book;
   }
 
-  makeChapter(projectName: string, bookName: string, name: string, directory: string): Chapter {
-    const chapter = new Chapter();
-    chapter.name = parseInt(name).toString();
-    const chapterFiles = fs.readdirSync(path.join(directory, projectName, bookName, name));
-    chapter.audioFiles = chapterFiles
-      .filter((file: string) => isValidAudioFile(file, DEFAULT_HEARTHIS_XML_FILE, fileFilters.audio[0].extensions))
-      .map((fileName: string) => path.join(directory, projectName, bookName, name, fileName));
-
-    chapter.textXmlFile = chapterFiles.find((file: string) => file === DEFAULT_HEARTHIS_XML_FILE);
-    if (chapter.textXmlFile)
-      chapter.textXmlFile = path.join(directory, projectName, bookName, name, chapter.textXmlFile);
-    chapter.fullPath = path.join(directory, projectName, bookName, name);
+  private makeChapter(
+    projectName: string,
+    bookName: string,
+    chapterName: string,
+    directory: string,
+    calculateAudioDuration = false
+  ): BKChapter {
+    const chapter: BKChapter = {
+      name: chapterName,
+      audio: {
+        files: [],
+        length: 0,
+      },
+      segments: [],
+    };
+    const sourceChapterDir = path.join(directory, projectName, bookName, chapterName);
+    const chapterFiles = fs.readdirSync(sourceChapterDir, 'utf8');
+    const audioFiles = chapterFiles.filter((file: string) => file.endsWith('.wav'));
```

Comment:
What if the extension is in caps? Perhaps use `file.toLowerCase().endsWith('.wav')`.

Reply:
Done